opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -466,11 +465,45 @@ public static void addObservation(Event e, JSONObject jsonObject) {
             } catch (JSONException e1) {
                 Timber.e(e1);
             }
+        } else if (AllConstants.GPS.equals(type)) {
+            createGpsObservation(e, jsonObject, value);
         } else {
             createObservation(e, jsonObject, value);
         }
     }
 
+    private static void createGpsObservation(Event e, JSONObject jsonObject, String value) {
```

Comment:
Any reason to save each observation separately? Wouldn't it be more compact and consistent to use the approach native radio widget takes of saving key-value pairs as part of the `obs` using https://github.com/OpenSRP/opensrp-client-core/blob/a6880cd75e0584bf39a08277fe9226f0078ec29f/opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java#L588?

Reply:
What happens on https://github.com/OpenSRP/opensrp-client-core/blob/a6880cd75e0584bf39a08277fe9226f0078ec29f/opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java#L597 ? Will it create separate obs with the key/value pair i provide?

Reply:
No it won't but is that necessary? This creates a block with the key-value pairs you pass as part of one `obs` e.g.

```
{
      "fieldType": "formsubmissionField",
      "fieldDataType": "text",
      "fieldCode": "travel_location",
      "parentCode": "",
      "values": [
        "Jakarta",
        "Depok",
        "Manado"
      ],
      "set": [],
      "formSubmissionField": "travel_location",
      "humanReadableValues": [],
      "keyValPairs": {
        "depok": "Depok",
        "manado": "Manado",
        "jakarta": "Jakarta"
      },
      "saveObsAsArray": false
 }
```

Reply:
the `keyValPairs` can contain lat,lng,alt etc. for instance.

Reply:
Yeah its necessary , I need them to be different obs.

Reply:
Could I ask why that is? Like what the use case is?